### PR TITLE
r-htmltable: package update

### PR DESCRIPTION
**Pull request by matteoguglielmi**

### CODE REVIEW

File: var/spack/repos/builtin/packages/r-htmltable/package.py [hunk 1]
```diff
@@ -35,13 +35,15 @@ class RHtmltable(RPackage):
...
 
+    version('1.11.2', '473e6d486e7714f8dd7f16a31480c896')
     version('1.9', '08c62c19e1ffe570e7d8fa57db5094b9')
 
     depends_on('r-stringr', type=('build', 'run'))
     depends_on('r-knitr', type=('build', 'run'))
     depends_on('r-magrittr', type=('build', 'run'))
     depends_on('r-checkmate', type=('build', 'run'))
     depends_on('r-htmlwidgets', type=('build', 'run'))
+    depends_on('r-rstudioapi', type=('build', 'run'), when="@1.11.2:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-htmltable/package.py]: Can you add the dependency version constraints? For example, `r-rstudioapi@0.6:`

Comment (matteoguglielmi) [reply var/spack/repos/builtin/packages/r-htmltable/package.py]: done (as specified in htmlTable/DESCRIPTION source file).

File: var/spack/repos/builtin/packages/r-htmltable/package.py [hunk 2]
```diff
@@ -35,13 +35,15 @@ class RHtmltable(RPackage):
...
     homepage = "https://CRAN.R-project.org/package=htmlTable"
-    url      = "https://cran.rstudio.com/src/contrib/htmlTable_1.9.tar.gz"
+    url      = "https://cran.rstudio.com/src/contrib/htmlTable_1.11.2.tar.gz"
     list_url = "https://cran.r-project.org/src/contrib/Archive/htmlTable"
 
+    version('1.11.2', '473e6d486e7714f8dd7f16a31480c896')
     version('1.9', '08c62c19e1ffe570e7d8fa57db5094b9')
 
     depends_on('r-stringr', type=('build', 'run'))
     depends_on('r-knitr', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-htmltable/package.py]: Can you specify the other dependency versions as well? See https://cran.r-project.org/web/packages/htmlTable/index.html

File: var/spack/repos/builtin/packages/r-htmltable/package.py [hunk 3]
```diff
@@ -35,13 +35,15 @@ class RHtmltable(RPackage):
...
     list_url = "https://cran.r-project.org/src/contrib/Archive/htmlTable"
 
+    version('1.11.2', '473e6d486e7714f8dd7f16a31480c896')
     version('1.9', '08c62c19e1ffe570e7d8fa57db5094b9')
 
     depends_on('r-stringr', type=('build', 'run'))
-    depends_on('r-knitr', type=('build', 'run'))
-    depends_on('r-magrittr', type=('build', 'run'))
+    depends_on('r-knitr@1.6:', type=('build', 'run'), when="@1.11.2:")
+    depends_on('r-magrittr@1.5:', type=('build', 'run'), when="@1.11.2:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-htmltable/package.py]: These dependencies are the same for version 1.9 as well, no need to add the `when` here.

File: var/spack/repos/builtin/packages/r-htmltable/package.py [hunk 4]
```diff
@@ -35,13 +35,15 @@ class RHtmltable(RPackage):
...
     depends_on('r-checkmate', type=('build', 'run'))
     depends_on('r-htmlwidgets', type=('build', 'run'))
+    depends_on('r-rstudioapi@0.6:', type=('build', 'run'), when="@1.11.2:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-htmltable/package.py]: This dependency was added in version 1.11.0, not 1.11.2.

File: var/spack/repos/builtin/packages/r-htmltable/package.py (hunk 3 above)
Comment (matteoguglielmi) [reply var/spack/repos/builtin/packages/r-htmltable/package.py]: Ok, 'whens' removed.

File: var/spack/repos/builtin/packages/r-htmltable/package.py (hunk 4 above)
Comment (matteoguglielmi) [reply var/spack/repos/builtin/packages/r-htmltable/package.py]: done.

will try to be more precise, this is my first set of packages for spack, I appreciate your guidelines.

**Pull request merged**